control-plane/subcommand/mesh-init/command.go

```diff
@@ -0,0 +1,283 @@
+		defer c.watcher.Stop()
+	}
+
+	state, err := c.watcher.State()
+	if err != nil {
+		c.logger.Error("Unable to get state from consul-server-connection-manager", "error", err)
+		return 1
+	}
+
+	consulClient, err := consul.NewClientFromConnMgrState(consulConfig, state)
+	if err != nil {
+		c.logger.Error("Unable to get client connection", "error", err)
+		return 1
+	}
+
+	if version.IsFIPS() {
+		// make sure we are also using FIPS Consul
+		var versionInfo map[string]interface{}
+		_, err := consulClient.Raw().Query("/v1/agent/version", versionInfo, nil)
+		if err != nil {
+			c.logger.Warn("This is a FIPS build of consul-k8s, which should be used with FIPS Consul. Unable to verify FIPS Consul while setting up Consul API client.")
+		}
+		if val, ok := versionInfo["FIPS"]; !ok || val == "" {
+			c.logger.Warn("This is a FIPS build of consul-k8s, which should be used with FIPS Consul. A non-FIPS version of Consul was detected.")
+		}
+	}
+
+	// todo (agentless): this should eventually be passed to consul-dataplane as a string so we don't need to write it to file.
+	if c.consul.UseTLS && c.consul.CACertPEM != "" {
+		if err = common.WriteFileWithPerms(constants.ConsulCAFile, c.consul.CACertPEM, 0444); err != nil {
+			c.logger.Error("error writing CA cert file", "error", err)
+			return 1
+		}
+	}
+
+	dc, err := consul.NewDataplaneServiceClient(c.watcher)
+	if err != nil {
+		c.logger.Error("failed to create resource client", "error", err)
+		return 1
+	}
+
+	var res pbdataplane.GetEnvoyBootstrapParamsResponse
+	if err := backoff.Retry(c.getBootstrapParams(dc, &res), backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), c.maxPollingAttempts)); err != nil {
+		c.logger.Error("Timed out waiting for bootstrap parameters", "error", err)
+		return 1
+	}
+
+	if c.flagRedirectTrafficConfig != "" {
+		err := c.applyTrafficRedirectionRules(res.GetBootstrapConfig()) // BootstrapConfig is always populated non-nil from the RPC
+		if err != nil {
+			c.logger.Error("error applying traffic redirection rules", "err", err)
+			return 1
+		}
+	}
+
+	c.logger.Info("Proxy initialization completed")
+	return 0
+}
+
+func (c *Command) validateFlags() error {
+	if c.flagProxyName == "" {
+		return errors.New("-proxy-name must be set")
+	}
+	return nil
+}
+
+func (c *Command) Synopsis() string { return synopsis }
+func (c *Command) Help() string {
+	c.once.Do(c.init)
+	return c.help
+}
+
+func (c *Command) getBootstrapParams(
+	client pbdataplane.DataplaneServiceClient,
+	response *pbdataplane.GetEnvoyBootstrapParamsResponse) backoff.Operation {
+
+	return func() error {
+		req := &pbdataplane.GetEnvoyBootstrapParamsRequest{
+			ProxyId:   c.flagProxyName,
+			Namespace: c.consul.Namespace,
+			Partition: c.consul.Partition,
+		}
+		res, err := client.GetEnvoyBootstrapParams(context.Background(), req)
+		if err != nil {
+			c.logger.Error("Unable to get bootstrap parameters", "error", err)
+			return err
+		}
+		*response = *res
```

Comment:
~ I think Goland warns about this due to the `DoNotCopy` pragma [embedded in proto-generated structs](https://stackoverflow.com/a/64183968), which I'm not terribly familiar w/ but I believe is there to protect some internal magic pointers from being copied.

I know the reason for this here is to escape the op closure and _might_ not cause issues given we just read the values and toss the result, but I wonder whether it's worth avoiding the direct copy, JIC and to avoid `go vet` complaining about it.

Reply:
👍 I swapped out the response struct with the bit of configuration I need (`pbmesh.BootstrapConfig`), and now the linter seems pleased.